migrate/list-convert.go

```diff
@@ -0,0 +1,296 @@
+	"github.com/minio/directpv/pkg/apis/directpv.min.io/types"
+	directpv "github.com/minio/directpv/pkg/apis/directpv.min.io/v1beta1"
+	"github.com/minio/directpv/pkg/consts"
+	directv1alpha1 "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1alpha1"
+	directv1beta1 "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta1"
+	directv1beta2 "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta2"
+	directv1beta3 "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta3"
+	directv1beta4 "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta4"
+	directcsi "github.com/minio/directpv/pkg/oldapis/direct.csi.min.io/v1beta5"
+	directcsiclient "github.com/minio/directpv/pkg/oldclient"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	MaxThreadCount = 200
+)
+
+func getKubeConfig() (*rest.Config, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return nil, err
+	}
+	kubeConfig := filepath.Join(home, ".kube", "config")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeConfig)
+	if err != nil {
+		if config, err = rest.InClusterConfig(); err != nil {
+			return nil, err
+		}
+	}
+	config.QPS = float32(MaxThreadCount / 2)
+	config.Burst = MaxThreadCount
+	return config, nil
+}
+
+func migrateVolumeCRD(oldVolumeList []directcsi.DirectCSIVolume) (newVolumeList []directpv.DirectPVVolume, theerror error) {
+	newVolumes := []directpv.DirectPVVolume{}
+	for _, OldVolume := range oldVolumeList {
+		NewVolume := directpv.DirectPVVolume{}
+
+		// TypeMeta fields
+		NewVolume.TypeMeta.Kind = consts.VolumeTypeMetaKind
+		NewVolume.TypeMeta.APIVersion = "directpv.min.io/v1beta1"
+
+		// ObjectMeta fields
+		NewVolume.ObjectMeta = OldVolume.ObjectMeta
+
+		// Status fields
+		NewVolume.Status.DataPath = OldVolume.Status.HostPath
+		NewVolume.Status.StagingTargetPath = OldVolume.Status.StagingPath
+		NewVolume.Status.TargetPath = OldVolume.Status.ContainerPath
+		NewVolume.Status.FSUUID = ""
+		NewVolume.Status.TotalCapacity = OldVolume.Status.TotalCapacity
+		NewVolume.Status.AvailableCapacity = OldVolume.Status.AvailableCapacity
+		NewVolume.Status.UsedCapacity = OldVolume.Status.UsedCapacity
+		NewVolume.Status.Conditions = OldVolume.Status.Conditions
+
+		// Append the new drive to the list
+		newVolumes = append(newVolumes, NewVolume)
+	}
+	return newVolumes, nil
+}
+
+func convertDirectCSITypeToString(x directcsi.AccessTier) string {
+	return string(x)
+}
+
+func convertStringToDirectPVType(x string) types.AccessTier {
+	return types.AccessTier(x)
+}
+
+func migrateDriveCRD(oldDriveList []directcsi.DirectCSIDrive) (newDriveList []directpv.DirectPVDrive, therror error) {
+
+	// Set of unique FSUUIDs:
+	setOfUniqueFSUUIDs := make(map[string]struct{})
+
+	newDrives := []directpv.DirectPVDrive{}
+	for _, OldDrive := range oldDriveList {
+		NewDrive := directpv.DirectPVDrive{}
```

Comment:
we can filter out the drives other than Ready and InUse

```go
if oldDrive.Status.DriveStatus != directcsi.DriveStatusReady && oldDrive.Status.DriveStatus != directcsi.DriveStatusInUse {
       continue
}
```

Reply:
Added already:

```
		// we can filter out the drives other than Ready and InUse
		if OldDrive.Status.DriveStatus != directcsi.DriveStatusReady && OldDrive.Status.DriveStatus != directcsi.DriveStatusInUse {
			continue // skip this iteration to filter out this state.
		}
```